public/app/pipeline-orchestrator.js

```
@@ -387,20 +394,35 @@ class PipelineOrchestrator {
   // ── Connect / disconnect an audio source ────────────────────────────────
-  /** @param {AudioNode} sourceNode */
+  /**
+   * Connect an audio source to the pipeline. Synchronous when the pipeline
+   * is already ready; otherwise defers the actual connection until readiness
+   * (or until timeout) so callers don't silently drop the source.
+   * @param {AudioNode} sourceNode
+   * @returns {Promise<boolean>} resolves true if connected, false otherwise.
+   */
   connectSource(sourceNode) {
     if (!this.workletNode) {
       console.warn('[Orchestrator] connectSource called before init()');
```

ESLint

[error] 'console' is not defined.

(no-undef)